src/app.rs

```diff
@@ -186,6 +197,74 @@ impl<'a, 'b> Application<'a, 'b> {
         self.engine.world.maintain();
     }
 
+    /// Input devices can sometimes generate a lot of motion events per frame, these are
+    /// useless as the extra precision is wasted and these events tend to overflow our
+    /// otherwise very adequate event buffers.  So this function removes and compresses redundant
+    /// events.
+    fn compress_events(vec: &mut Vec<Event>, new_event: Event) {
+        match new_event {
+            Event::WindowEvent { ref event, .. } => match event {
+                &WindowEvent::MouseMoved { .. } => for mut stored_event in vec.iter_mut() {
+                    if let &mut Event::WindowEvent {
+                        event: WindowEvent::MouseMoved { .. },
```

Comment:
There is `device_id` as well.

Reply:
Gotta filter by mouse pointer so we can support this feature: https://youtu.be/UiLSiqyDf4Y :P 

I don't feel it's necessary to filter MouseMoved by device the event came from.